isobus/include/isobus/isobus/can_network_manager.hpp

```diff
@@ -137,6 +137,17 @@ namespace isobus
 		/// @param[in] controlFunction The control function that was created
 		void on_control_function_created(std::shared_ptr<ControlFunction> controlFunction, CANLibBadge<ControlFunction>);
 
+		/// @brief Use this to get a callback when a control function goes online or offline.
+		/// This could be useful if you want event driven notifications for when your partners are disconnected from the bus.
+		/// @param[in] controlFunction The control function you want callbacks for
+		/// @param[in] callback The callback you want to be called when the specified control function changes state
+		void add_control_function_status_change_callback(std::shared_ptr<ControlFunction> controlFunction, ControlFunctionStateCallback callback);
```

Comment:
Hmm, wouldn't it make more sense to just register a callback for all controlfunctions instead of binding it to a single one. It allows for greater flexability in use cases, and doesn't affect performance I think

Reply:
Hm, I didn't think about that. That would probably be even more performant. I like it - I will change it.